#161255325  Article tags should be present in all relevant views

njeri-ngigi opened the pull request:
#### What does this PR do?
- Adds a tagging component that can be passed to any view 
- Adds tests for this component
#### Description of tasks to be completed
>  Article tags should be present in all relevant views

**Why this is important**
Tagging refers to the metadata that is assigned to a piece of content by the content creator and the readers/users of the content. Tagging allows for organization and or categorization of content

**As a User**
I want to be able to view various article tags to enable easy filtering and navigation to articles of interest 

**Acceptance Criteria**
Scenario:
Only **authenticated** users should be able to tag articles .

#### Relevant PT stories
[Article tags should be present in all relevant views](https://www.pivotaltracker.com/story/show/161255325)

#### Screenshots:
![image](https://user-images.githubusercontent.com/28973383/49945926-f0ce4b80-fefe-11e8-9b92-0b41c0529d7a.png)

njeri-ngigi committed:
(Feature #161255322): Pagination
- Add pagination component
[Feature #161255322]

njeri-ngigi committed:
(Feature #161255322): Pagination
- Add pagination component
[Feature #161255322]

njeri-ngigi committed:
(Feature #161255322): Pagination
- Add action for fetching user stories
[Feature #161255322]

njeri-ngigi committed:
(Feature #161255322): Pagination
- Add action for fetching user stories
[Feature #161255322]

njeri-ngigi committed:
(Feature #161255322): Pagination
- Add pagination for user articles in the profile
[Feature #161255322]

njeri-ngigi committed:
(Feature #161255322): Pagination
- Add pagination for user articles in the profile
[Feature #161255322]

njeri-ngigi committed:
(Feature #161255322): Pagination
- Adjust page links css
[Feature #161255322]

njeri-ngigi committed:
(Feature #161255322): Pagination
- Adjust page links css
[Feature #161255322]

3N61N33R committed:
feat(rating): rating article functionality

- user rates article
- gets article average rating

[Finishes #161255320]

3N61N33R committed:
feat(rating): rating article functionality

- user rates article
- gets article average rating

[Finishes #161255320]

njeri-ngigi committed:
(Feature #161255325): Add tagging on all relevant views:
- Add a tagging component
- Add method in articleActions to fetch tags
[Feature #161255325]

njeri-ngigi committed:
(Feature #161255325): Add tagging on all relevant views:
- Add a tagging component
- Add method in articleActions to fetch tags
[Feature #161255325]

njeri-ngigi committed:
(Feature #161255325): Add tagging on all relevant views:
- Add a tagging component
- Add method in articleActions to fetch tags
[Feature #161255325]

njeri-ngigi committed:
(Feature #161255325): Add tagging on all relevant views:
- Add a tagging component
- Add method in articleActions to fetch tags
[Feature #161255325]

njeri-ngigi committed:
(Feature #161255325): Add tagging on all relevant views:
- Add a tagging component
- Add method in articleActions to fetch tags
[Feature #161255325]

njeri-ngigi committed:
(Feature #161255325): Add tagging on all relevant views:
- Add a tagging component
- Add method in articleActions to fetch tags
[Feature #161255325]

njeri-ngigi committed:
(Feature #161255325): Add tagging on all relevant views:
- Add a tagging component
- Add method in articleActions to fetch tags
[Feature #161255325]

njeri-ngigi committed:
(Feature #161255325): Add tagging on all relevant views:
- Add a tagging component
- Add method in articleActions to fetch tags
[Feature #161255325]

njeri-ngigi committed:
Merge branch 'develop' into ft-pagination-161255322

AnguleMathias committed:
Feature(Articles CRUD): Users should be able to Create, Read, Delete and Update Articles

- CRUD functionality for articles

[Feature #161255319]

AnguleMathias committed:
Feature(Articles CRUD): Users should be able to Create, Read, Delete and Update Articles

- tests for CRUD functionality for articles

[Feature #161255319]

3N61N33R committed:
feat(rating): rating articles

- add tests

[finishes #161255320]

AnguleMathias reviewed (changes requested):
Good work on feature implementation. Work on fixing code climate issue and improve on test coverage.

Commit:
Feature(Articles CRUD): Users should be able to Create, Read, Delete and Update Articles

- Refactor code

[Feature #161255319]

Commit:
Feature(Articles CRUD): Users should be able to Create, Read, Delete and Update Articles

- Refactor code

[Feature #161255319]

AnguleMathias committed:
Merge branch 'ft-articles-161255319' of https://github.com/andela/ah-codeofduty-frontend into ft-articles-161255319

AnguleMathias committed:
Feature(Articles CRUD): Users should be able to Create, Read, Delete and Update Articles

- Modified edit article form
- Add article created date as a string
- Add article reading time
- Modified edit and delete button to appear only to article author

[Feature #161255319]

Commit:
(Feature #161255322): Pagination tests
- Add test to check for correct mounting of paginate component
[Feature #161255322]

Commit:
(Feature #161255322): Pagination tests
- Add test to check for correct mounting of paginate component
[Feature #161255322]

AnguleMathias committed:
Feature(Articles CRUD): Users should be able to Create, Read, Delete and Update Articles

- UI feedback implementation

[Feature #161255319]

AnguleMathias committed:
Feature(Articles CRUD): Users should be able to Create, Read, Delete and Update Articles

- UI feedback implementation

[Feature #161255319]

njeri-ngigi committed:
Merge pull request #19 from andela/ft-articles-161255319

#161255319 Users should be able to create articles

Commit:
(Feature #161255322): Pagination tests
- Add test to check for correct mounting of paginate component
[Feature #161255322]

3N61N33R committed:
Merge branch 'develop' into ft-rate-articles-161255320

Commit:
(Feature #161255322): Pagination
- Add pagination component to articles page
- Add pagination component to profile pages to show articles
- Adjust article reducer to have pageCount stored in state
[Feature #161255322]

njeri-ngigi committed:
Merge pull request #16 from andela/ft-pagination-161255322

#161255322 Pagination support for articles

3N61N33R committed:
feat(rating): rating article functionality

-add ratings to article

[feature #161255320]

3N61N33R committed:
Merge branch 'develop' into ft-rate-articles-161255320

3N61N33R committed:
mend

maxwellgithinji committed:
#161255321 Users should be able to comment on articles (#21)

* Feature(Articles CRUD): Users should be able to Create, Read, Delete and Update Articles

- CRUD functionality for articles

[Feature #161255319]

* Feature(Articles CRUD): Users should be able to Create, Read, Delete and Update Articles

- tests for CRUD functionality for articles

[Feature #161255319]

* Feature(Articles CRUD): Users should be able to Create, Read, Delete and Update Articles

- Refactor code

[Feature #161255319]

* Feature(Articles CRUD): Users should be able to Create, Read, Delete and Update Articles

- Refactor code

[Feature #161255319]

* Feature(Articles CRUD): Users should be able to Create, Read, Delete and Update Articles

- Modified edit article form
- Add article created date as a string
- Add article reading time
- Modified edit and delete button to appear only to article author

[Feature #161255319]

* Feature(Articles CRUD): Users should be able to Create, Read, Delete and Update Articles

- UI feedback implementation

[Feature #161255319]

* Feature Users should be able to comment on articles:
 - ensure user is able to make a comment(s) on articles, both authored by them and those not authored by them
 - ensure Only authenticated users should be able to comments on articles.
 - ensure Successive comments should be threaded
 - ensure Timestamped comments.
[Feature #161255321]

* Feature(Articles CRUD): Users should be able to Create, Read, Delete and Update Articles

- UI feedback implementation

[Feature #161255319]

* Feature Users should be able to comment on articles:
 - ensure user is able to make a comment(s) on articles, both authored by them and those not authored by them
 - ensure Only authenticated users should be able to comments on articles.
 - ensure Successive comments should be threaded
 - ensure Timestamped comments.
[Feature #161255321]

* Feature  Users should be able to comment on articles:
- ensure to integrate comments with articles
[Feature #161255321]

3N61N33R committed:
mend

3N61N33R committed:
Merge branch 'develop' into ft-rate-articles-161255320

3N61N33R committed:
mend

njeri-ngigi committed:
Merge pull request #20 from andela/ft-rate-articles-161255320

#161255320 Users should be able to rate articles

njeri-ngigi committed:
(Feature #161255325): Tags should be present in relevant views
- add input to add or edit tags
- Add handle to fetch filter by tags
- Update signup and login urls to have a common base url
[Feature #161255325]

njeri-ngigi committed:
(Feature #161255325): Tags should be present in relevant views
- add input to add or edit tags
- Add handle to fetch filter by tags
- Update signup and login urls to have a common base url
[Feature #161255325]

njeri-ngigi committed:
(Feature #161255325): Tags should be present in relevant views
- add input to add or edit tags
- Add handle to fetch filter by tags
- Update signup and login urls to have a common base url
[Feature #161255325]

njeri-ngigi committed:
(Feature #161255325): Tags should be present in relevant views
- add input to add or edit tags
- Add handle to fetch filter by tags
- Update signup and login urls to have a common base url
[Feature #161255325]

njeri-ngigi committed:
(Feature #161255325): Tags should be present in relevant views
- add tests for the tag component
[Feature #161255325]

njeri-ngigi committed:
(Feature #161255325): Tags should be present in relevant views
- add tests for the tag component
[Feature #161255325]

njeri-ngigi committed:
(Feature #161255325): Tags should be present in relevant views
- add tests for the tag component
[Feature #161255325]

njeri-ngigi committed:
(Feature #161255325): Tags should be present in relevant views
- add tests for the tag component
[Feature #161255325]

njeri-ngigi committed:
(Feature #161255325): Tags should be present in relevant views
- add tests for the tag component
[Feature #161255325]

njeri-ngigi committed:
(Feature #161255325): Tags should be present in relevant views
- add css outline: none property on input and button elements
[Feature #161255325]

njeri-ngigi committed:
(Feature #161255325): Tags should be present in relevant views
- add css outline: none property on input and button elements
[Feature #161255325]

njeri-ngigi committed:
(Feature #161255325): Tags should be present in relevant views
- add css outline: none property on input and button elements
[Feature #161255325]

njeri-ngigi committed:
(Feature #161255325): Tagging Bug Fix
- Added rating feature deleted during merging
- Fixed bug in registerUser method
[Feature #161255325]

njeri-ngigi committed:
(Feature #161255325): Tagging Bug Fix
- Added rating feature deleted during merging
- Fixed bug in registerUser method
[Feature #161255325]

njeri-ngigi committed:
(Feature #161255325): Tagging Bug Fix
- Added rating feature deleted during merging
- Fixed bug in registerUser method
[Feature #161255325]

njeri-ngigi committed:
(Feature #161255325): Tagging Bug Fix
- Added rating feature deleted during merging
- Fixed bug in registerUser method
[Feature #161255325]

Commit:
(feature #161255325): Tagging

-fix loginUser

[Feature #161255325]

Commit:
(feature #161255325): Tagging

-fix loginUser

[Feature #161255325]

Commit:
(feature #161255325): Tagging

-fix loginUser

[Feature #161255325]

KITHU reviewed (changes requested):
kindly remove complexity: hint reduce number of lines in your render method

KITHU committed:
161255315-ft-reset-password  (#18)

* (Feature #161255322): Pagination
- Add pagination component
[Feature #161255322]

* (Feature #161255322): Pagination
- Add action for fetching user stories
[Feature #161255322]

* (Feature #161255322): Pagination
- Add pagination for user articles in the profile
[Feature #161255322]

* (Feature #161255322): Pagination
- Adjust page links css
[Feature #161255322]

* feat(rating): rating article functionality

- user rates article
- gets article average rating

[Finishes #161255320]

* feat(reset password): reset password after forgeting

- A user should be presented with a button and or link to reset their passwords on the SignUp/Register component

[Delivers #161255315]

* feat(forgot password):forgot password
- added codeclimate yaml file
[Delivers #161255315]

* feat(forgot password):forgot password
- added codeclimate yaml file
[Delivers #161255315]

* Feature(Articles CRUD): Users should be able to Create, Read, Delete and Update Articles

- CRUD functionality for articles

[Feature #161255319]

* Feature(Articles CRUD): Users should be able to Create, Read, Delete and Update Articles

- tests for CRUD functionality for articles

[Feature #161255319]

* resolve conflict

* resolve conflict

* feat(rating): rating article functionality

- user rates article
- gets article average rating

[Finishes #161255320]

* Feature(Articles CRUD): Users should be able to Create, Read, Delete and Update Articles

- Refactor code

[Feature #161255319]

* Feature(Articles CRUD): Users should be able to Create, Read, Delete and Update Articles

- Refactor code

[Feature #161255319]

* Feature(Articles CRUD): Users should be able to Create, Read, Delete and Update Articles

- Modified edit article form
- Add article created date as a string
- Add article reading time
- Modified edit and delete button to appear only to article author

[Feature #161255319]

* (Feature #161255322): Pagination tests
- Add test to check for correct mounting of paginate component
[Feature #161255322]

* Feature(Articles CRUD): Users should be able to Create, Read, Delete and Update Articles

- UI feedback implementation

[Feature #161255319]

* Feature(Articles CRUD): Users should be able to Create, Read, Delete and Update Articles

- UI feedback implementation

[Feature #161255319]

* (Feature #161255322): Pagination
- Add pagination component
[Feature #161255322]

* (Feature #161255322): Pagination
- Add action for fetching user stories
[Feature #161255322]

* (Feature #161255322): Pagination
- Add pagination for user articles in the profile
[Feature #161255322]

* (Feature #161255322): Pagination
- Adjust page links css
[Feature #161255322]

* (Feature #161255322): Pagination tests
- Add test to check for correct mounting of paginate component
[Feature #161255322]

* (Feature #161255322): Pagination tests
- Add test to check for correct mounting of paginate component
[Feature #161255322]

* feat(rating): rating article functionality

-add ratings to article

[feature #161255320]

* #161255321 Users should be able to comment on articles (#21)

* Feature(Articles CRUD): Users should be able to Create, Read, Delete and Update Articles

- CRUD functionality for articles

[Feature #161255319]

* Feature(Articles CRUD): Users should be able to Create, Read, Delete and Update Articles

- tests for CRUD functionality for articles

[Feature #161255319]

* Feature(Articles CRUD): Users should be able to Create, Read, Delete and Update Articles

- Refactor code

[Feature #161255319]

* Feature(Articles CRUD): Users should be able to Create, Read, Delete and Update Articles

- Refactor code

[Feature #161255319]

* Feature(Articles CRUD): Users should be able to Create, Read, Delete and Update Articles

- Modified edit article form
- Add article created date as a string
- Add article reading time
- Modified edit and delete button to appear only to article author

[Feature #161255319]

* Feature(Articles CRUD): Users should be able to Create, Read, Delete and Update Articles

- UI feedback implementation

[Feature #161255319]

* Feature Users should be able to comment on articles:
 - ensure user is able to make a comment(s) on articles, both authored by them and those not authored by them
 - ensure Only authenticated users should be able to comments on articles.
 - ensure Successive comments should be threaded
 - ensure Timestamped comments.
[Feature #161255321]

* Feature(Articles CRUD): Users should be able to Create, Read, Delete and Update Articles

- UI feedback implementation

[Feature #161255319]

* Feature Users should be able to comment on articles:
 - ensure user is able to make a comment(s) on articles, both authored by them and those not authored by them
 - ensure Only authenticated users should be able to comments on articles.
 - ensure Successive comments should be threaded
 - ensure Timestamped comments.
[Feature #161255321]

* Feature  Users should be able to comment on articles:
- ensure to integrate comments with articles
[Feature #161255321]

* mend

* mend

* feat(reset password): reset password after forgeting

- A user should be presented with a button and or link to reset their passwords on the SignUp/Register component

[Delivers #161255315]

* feat(forgot password):forgot password
- added codeclimate yaml file
[Delivers #161255315]

* feat(forgot password):forgot password
- added codeclimate yaml file
[Delivers #161255315]

* resolve conflict

* resolve conflict

njeri-ngigi committed:
(Feature #161255325): Add Tagging to all relevant views
- Adjust css properties of the Tags component (padding)
[Feature #161255325]

njeri-ngigi committed:
(Feature #161255325): Add Tagging to all relevant views
- Adjust css properties of the Tags component (padding)
[Feature #161255325]

njeri-ngigi committed:
(Feature #161255325): Tagging
- Fix merge conflicts after rebasing
[Feature #161255325]

KITHU committed:
161255324-Ft like and dislike articles (#22)

* feat(like): like unlike article

- user can like an article
- users can dislike articles

[Delivers #161255324]

* feat(like): like unlike article

- user can like an article
- users can dislike articles

[Delivers #161255324]

njeri-ngigi committed:
(Feature #161255325): Tagging
- Add a tags reducer
[Feature #161255325]

njeri-ngigi committed:
(Feature #161255325): Tagging
- Add a tags reducer
[Feature #161255325]

njeri-ngigi committed:
(Feature #161255325): Tagging
- Adjust css properties of like and unlike icons
[Feature #161255325]

njeri-ngigi committed:
Feature #161255325: Tagging
- Fix merge conflicts from remote
[Feature #161255325]
